From 854f6157fa981119d876f62fd78898600c3235e8 Mon Sep 17 00:00:00 2001
From: link07 <link07@users.noreply.github.com>
Date: Sat, 24 Feb 2018 18:30:42 -0700
Subject: [PATCH] Bitcoin->Lightcoin conversion for bash-completion and
 bitcoind


diff --git a/contrib/bitcoin-cli.bash-completion b/contrib/bitcoin-cli.bash-completion
index 732981fe7..499e007f2 100644
--- a/contrib/bitcoin-cli.bash-completion
+++ b/contrib/bitcoin-cli.bash-completion
@@ -4,7 +4,7 @@
 # file COPYING or http://www.opensource.org/licenses/mit-license.php.
 
 # call $bitcoin-cli for RPC
-_bitcoin_rpc() {
+_litecoin_rpc() {
     # determine already specified args necessary for RPC
     local rpcargs=()
     for i in ${COMP_LINE}; do
@@ -14,23 +14,23 @@ _bitcoin_rpc() {
                 ;;
         esac
     done
-    $bitcoin_cli "${rpcargs[@]}" "$@"
+    $litecoin_cli "${rpcargs[@]}" "$@"
 }
 
 # Add wallet accounts to COMPREPLY
-_bitcoin_accounts() {
+_litecoin_accounts() {
     local accounts
-    accounts=$(_bitcoin_rpc listaccounts | awk -F '"' '{ print $2 }')
+    accounts=$(_litecoin_rpc listaccounts | awk -F '"' '{ print $2 }')
     COMPREPLY=( "${COMPREPLY[@]}" $( compgen -W "$accounts" -- "$cur" ) )
 }
 
-_bitcoin_cli() {
+_litecoin_cli() {
     local cur prev words=() cword
-    local bitcoin_cli
+    local litecoin_cli
 
     # save and use original argument to invoke bitcoin-cli for -help, help and RPC
     # as bitcoin-cli might not be in $PATH
-    bitcoin_cli="$1"
+    litecoin_cli="$1"
 
     COMPREPLY=()
     _get_comp_words_by_ref -n = cur prev words cword
@@ -60,7 +60,7 @@ _bitcoin_cli() {
     if ((cword > 3)); then
         case ${words[cword-3]} in
             addmultisigaddress)
-                _bitcoin_accounts
+                _litecoin_accounts
                 return 0
                 ;;
             getbalance|gettxout|importaddress|importpubkey|importprivkey|listreceivedbyaccount|listreceivedbyaddress|listsinceblock)
@@ -85,7 +85,7 @@ _bitcoin_cli() {
                 return 0
                 ;;
             move|setaccount)
-                _bitcoin_accounts
+                _litecoin_accounts
                 return 0
                 ;;
         esac
@@ -101,7 +101,7 @@ _bitcoin_cli() {
             return 0
             ;;
         getaccountaddress|getaddressesbyaccount|getbalance|getnewaddress|getreceivedbyaccount|listtransactions|move|sendfrom|sendmany)
-            _bitcoin_accounts
+            _litecoin_accounts
             return 0
             ;;
     esac
@@ -125,12 +125,12 @@ _bitcoin_cli() {
 
             # only parse -help if senseful
             if [[ -z "$cur" || "$cur" =~ ^- ]]; then
-                helpopts=$($bitcoin_cli -help 2>&1 | awk '$1 ~ /^-/ { sub(/=.*/, "="); print $1 }' )
+                helpopts=$($litecoin_cli -help 2>&1 | awk '$1 ~ /^-/ { sub(/=.*/, "="); print $1 }' )
             fi
 
             # only parse help if senseful
             if [[ -z "$cur" || "$cur" =~ ^[a-z] ]]; then
-                commands=$(_bitcoin_rpc help 2>/dev/null | awk '$1 ~ /^[a-z]/ { print $1; }')
+                commands=$(_litecoin_rpc help 2>/dev/null | awk '$1 ~ /^[a-z]/ { print $1; }')
             fi
 
             COMPREPLY=( $( compgen -W "$helpopts $commands" -- "$cur" ) )
@@ -143,7 +143,7 @@ _bitcoin_cli() {
             ;;
     esac
 } &&
-complete -F _bitcoin_cli bitcoin-cli
+complete -F _litecoin_cli litecoin-cli
 
 # Local variables:
 # mode: shell-script
diff --git a/contrib/bitcoin-tx.bash-completion b/contrib/bitcoin-tx.bash-completion
index a83d2979e..e8b38e0f6 100644
--- a/contrib/bitcoin-tx.bash-completion
+++ b/contrib/bitcoin-tx.bash-completion
@@ -3,13 +3,13 @@
 # Distributed under the MIT software license, see the accompanying
 # file COPYING or http://www.opensource.org/licenses/mit-license.php.
 
-_bitcoin_tx() {
+_litecoin_tx() {
     local cur prev words=() cword
-    local bitcoin_tx
+    local litecoin_tx
 
-    # save and use original argument to invoke bitcoin-tx for -help
+    # save and use original argument to invoke litecoin-tx for -help
     # it might not be in $PATH
-    bitcoin_tx="$1"
+    litecoin_tx="$1"
 
     COMPREPLY=()
     _get_comp_words_by_ref -n =: cur prev words cword
@@ -27,15 +27,15 @@ _bitcoin_tx() {
 
     if [[ "$cword" == 1 || ( "$prev" != "-create" && "$prev" == -* ) ]]; then
         # only options (or an uncompletable hex-string) allowed
-        # parse bitcoin-tx -help for options
+        # parse litecoin-tx -help for options
         local helpopts
-        helpopts=$($bitcoin_tx -help | sed -e '/^  -/ p' -e d )
+        helpopts=$($litecoin_tx -help | sed -e '/^  -/ p' -e d )
         COMPREPLY=( $( compgen -W "$helpopts" -- "$cur" ) )
     else
         # only commands are allowed
         # parse -help for commands
         local helpcmds
-        helpcmds=$($bitcoin_tx -help | sed -e '1,/Commands:/d' -e 's/=.*/=/' -e '/^  [a-z]/ p' -e d )
+        helpcmds=$($litecoin_tx -help | sed -e '1,/Commands:/d' -e 's/=.*/=/' -e '/^  [a-z]/ p' -e d )
         COMPREPLY=( $( compgen -W "$helpcmds" -- "$cur" ) )
     fi
 
@@ -46,7 +46,7 @@ _bitcoin_tx() {
 
     return 0
 } &&
-complete -F _bitcoin_tx bitcoin-tx
+complete -F _litecoin_tx litecoin-tx
 
 # Local variables:
 # mode: shell-script
diff --git a/contrib/bitcoind.bash-completion b/contrib/bitcoind.bash-completion
index cccd4bde0..f35c7e016 100644
--- a/contrib/bitcoind.bash-completion
+++ b/contrib/bitcoind.bash-completion
@@ -1,15 +1,15 @@
-# bash programmable completion for bitcoind(1) and bitcoin-qt(1)
-# Copyright (c) 2012-2016 The Bitcoin Core developers
+# bash programmable completion for litecoind(1) and litecoin-qt(1)
+# Copyright (c) 2012-2016 The litecoin Core developers
 # Distributed under the MIT software license, see the accompanying
 # file COPYING or http://www.opensource.org/licenses/mit-license.php.

-_bitcoind() {
+_litecoind() {
     local cur prev words=() cword
-    local bitcoind
+    local litecoind

-    # save and use original argument to invoke bitcoind for -help
+    # save and use original argument to invoke litecoind for -help
     # it might not be in $PATH
-    bitcoind="$1"
+    litecoind="$1"

     COMPREPLY=()
     _get_comp_words_by_ref -n = cur prev words cword
@@ -33,7 +33,7 @@ _bitcoind() {
             # only parse -help if sensible
             if [[ -z "$cur" || "$cur" =~ ^- ]]; then
                 local helpopts
-                helpopts=$($bitcoind -help 2>&1 | awk '$1 ~ /^-/ { sub(/=.*/, "="); print $1 }' )
+                helpopts=$($litecoind -help 2>&1 | awk '$1 ~ /^-/ { sub(/=.*/, "="); print $1 }' )
                 COMPREPLY=( $( compgen -W "$helpopts" -- "$cur" ) )
             fi

@@ -45,7 +45,7 @@ _bitcoind() {
             ;;
     esac
 } &&
-complete -F _bitcoind bitcoind bitcoin-qt
+complete -F _litecoind litecoind litecoin-qt

 # Local variables:
 # mode: shell-script
diff --git a/contrib/init/bitcoind.service b/contrib/init/bitcoind.service
index ee113d761..c05fea141 100644
--- a/contrib/init/bitcoind.service
+++ b/contrib/init/bitcoind.service
@@ -1,23 +1,23 @@
 # It is not recommended to modify this file in-place, because it will
 # be overwritten during package upgrades. If you want to add further
 # options or overwrite existing ones then use
-# $ systemctl edit bitcoind.service
+# $ systemctl edit litecoind.service
 # See "man systemd.service" for details.

 # Note that almost all daemon options could be specified in
-# /etc/bitcoin/bitcoin.conf
+# /etc/litecoin/litecoin.conf

 [Unit]
-Description=Bitcoin daemon
+Description=litecoin daemon
 After=network.target

 [Service]
-ExecStart=/usr/bin/bitcoind -daemon -conf=/etc/bitcoin/bitcoin.conf -pid=/run/bitcoind/bitcoind.pid
-# Creates /run/bitcoind owned by bitcoin
-RuntimeDirectory=bitcoind
-User=bitcoin
+ExecStart=/usr/bin/litecoind -daemon -conf=/etc/litecoin/litecoin.conf -pid=/run/litecoind/litecoind.pid
+# Creates /run/litecoind owned by litecoin
+RuntimeDirectory=litecoind
+User=litecoin
 Type=forking
-PIDFile=/run/bitcoind/bitcoind.pid
+PIDFile=/run/litecoind/litecoind.pid
 Restart=on-failure
 PrivateTmp=true

--
2.17.0
